### ISSUE TITLE
Task Create zooker acl

### ISSUE THREAD

**Issue by kenjones-cisco** [CiscoCloud/microservices-infrastructure#154]
I thought it was an issue regarding the port 2181 not being open, but after including within Security Group, the issue has continued.

```
TASK: [mesos | create zookeeper acl] ******************************************
failed: [cchub-dc1-01] => {"changed": true, "cmd": "docker run --rm -e ZK_AUTH=super:/hp44hDBHEr8rODguwQkwz0i9Tra+G6+tgwzxO2iLhO4 -e ZNODE=/mesos -e ZNODE_ACL=world:anyone:r,digest:mesos:y++Ao12LgfFaQVuM+TQqx0gq0LE=:cdraw -e ZK_SERVER=zookeeper.service.consul:2181 --entrypoint /usr/local/bin/setacl.sh asteris/zookeeper:latest", "delta": "0:00:07.992321", "end": "2015-04-07 21:29:28.693678", "rc": 1, "start": "2015-04-07 21:29:20.701357", "warnings": []}
stderr: Exception in thread "main" org.apache.zookeeper.KeeperException$ConnectionLossException: KeeperErrorCode = ConnectionLoss for /mesos
        at org.apache.zookeeper.KeeperException.create(KeeperException.java:99)
        at org.apache.zookeeper.KeeperException.create(KeeperException.java:51)
        at org.apache.zookeeper.ZooKeeper.create(ZooKeeper.java:783)
        at org.apache.zookeeper.ZooKeeperMain.processZKCmd(ZooKeeperMain.java:695)
        at org.apache.zookeeper.ZooKeeperMain.processCmd(ZooKeeperMain.java:588)
        at org.apache.zookeeper.ZooKeeperMain.executeLine(ZooKeeperMain.java:360)
        at org.apache.zookeeper.ZooKeeperMain.run(ZooKeeperMain.java:323)
        at org.apache.zookeeper.ZooKeeperMain.main(ZooKeeperMain.java:282)
stdout: Connecting to zookeeper.service.consul:2181
2015-04-07 22:29:21,635 [myid:] - INFO  [main:Environment@100] - Client environment:zookeeper.version=3.4.6-1569965, built on 02/20/2014 09:09 GMT
2015-04-07 22:29:21,638 [myid:] - INFO  [main:Environment@100] - Client environment:host.name=1d33a0001719
2015-04-07 22:29:21,639 [myid:] - INFO  [main:Environment@100] - Client environment:java.version=1.7.0_71
2015-04-07 22:29:21,641 [myid:] - INFO  [main:Environment@100] - Client environment:java.vendor=Oracle Corporation
2015-04-07 22:29:21,641 [myid:] - INFO  [main:Environment@100] - Client environment:java.home=/usr/lib/jvm/java-1.7.0-openjdk-1.7.0.71-2.5.3.1.el7_0.x86_64/jre
2015-04-07 22:29:21,642 [myid:] - INFO  [main:Environment@100] - Client environment:java.class.path=/opt/zookeeper/bin/../build/classes:/opt/zookeeper/bin/../build/lib/*.jar:/opt/zookeeper/bin/../lib/slf4j-log4j12-1.6.1.jar:/opt/zookeeper/bin/../lib/slf4j-api-1.6.1.jar:/opt/zookeeper/bin/../lib/netty-3.7.0.Final.jar:/opt/zookeeper/bin/../lib/log4j-1.2.16.jar:/opt/zookeeper/bin/../lib/jline-0.9.94.jar:/opt/zookeeper/bin/../zookeeper-3.4.6.jar:/opt/zookeeper/bin/../src/java/lib/*.jar:/opt/zookeeper/bin/../conf:
2015-04-07 22:29:21,642 [myid:] - INFO  [main:Environment@100] - Client environment:java.library.path=/usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib
2015-04-07 22:29:21,642 [myid:] - INFO  [main:Environment@100] - Client environment:java.io.tmpdir=/tmp
2015-04-07 22:29:21,642 [myid:] - INFO  [main:Environment@100] - Client environment:java.compiler=<NA>
2015-04-07 22:29:21,642 [myid:] - INFO  [main:Environment@100] - Client environment:os.name=Linux
2015-04-07 22:29:21,642 [myid:] - INFO  [main:Environment@100] - Client environment:os.arch=amd64
2015-04-07 22:29:21,642 [myid:] - INFO  [main:Environment@100] - Client environment:os.version=3.10.0-123.el7.x86_64
2015-04-07 22:29:21,642 [myid:] - INFO  [main:Environment@100] - Client environment:user.name=root
2015-04-07 22:29:21,642 [myid:] - INFO  [main:Environment@100] - Client environment:user.home=/root
2015-04-07 22:29:21,642 [myid:] - INFO  [main:Environment@100] - Client environment:user.dir=/opt/zookeeper
2015-04-07 22:29:21,644 [myid:] - INFO  [main:ZooKeeper@438] - Initiating client connection, connectString=zookeeper.service.consul:2181 sessionTimeout=30000 watcher=org.apache.zookeeper.ZooKeeperMain$MyWatcher@7718bb9c
Welcome to ZooKeeper!
JLine support is enabled
2015-04-07 22:29:21,708 [myid:] - INFO  [main-SendThread(10.7.7.39:2181):ClientCnxn$SendThread@975] - Opening socket connection to server 10.7.7.39/10.7.7.39:2181. Will not attempt to authenticate using SASL (unknown error)
2015-04-07 22:29:21,720 [myid:] - INFO  [main-SendThread(10.7.7.39:2181):ClientCnxn$SendThread@852] - Socket connection established to 10.7.7.39/10.7.7.39:2181, initiating session
2015-04-07 22:29:21,727 [myid:] - INFO  [main-SendThread(10.7.7.39:2181):ClientCnxn$SendThread@1098] - Unable to read additional data from server sessionid 0x0, likely server has closed socket, closing socket connection and attempting reconnect
[zk: zookeeper.service.consul:2181(CONNECTING) 0] addauth digest super:/hp44hDBHEr8rODguwQkwz0i9Tra+G6+tgwzxO2iLhO4
[zk: zookeeper.service.consul:2181(CONNECTING) 1]
[zk: zookeeper.service.consul:2181(CONNECTING) 1] create /mesos ""
```

**Comment by stevendborrelli** [comment]
Can you also try opening ports 2888 and 3888? Those are also ports that zookeeper uses to communicate between nodes.

I'll mark this for the docs. I think the easiest thing to do is to create an openstack security role where all ports are trusted within the security group.

**Comment by kenjones-cisco** [comment]
I created a rule on my Security group "cluster", where all ports are trusted within the security group. And that has resolved the problem, and my playbook has completed successfully.

**Comment by stevendborrelli** [comment]
@kenjones-cisco thanks for all your debugging and feedback. It will help the project become that more usable for other people. I'm going to leave this issue open until we update the docs.

**Comment by stevendborrelli** [comment]
Fixed via #155 and #165.
